connector/session.py

```diff
@@ -202,6 +202,11 @@ def rollback(self):
 
     def close(self):
         """ Close the cursor """
+        try:
+            # Try to recompute fields before closing the cursor
+            self.env['ir.model'].recompute()
+        except:
+            self.rollback()
```

Comment:
This means we would remove the todo instead of calculating them. It should come to the same result since they are stored fields, and is probably cleaner than trying to calculate them.

But shouldn't we keep this cleaning in ```ConnectorSession.close()``` in case we get the error from somewhere else ?  

Or maybe we could modify ```ConnectorSessionHandler.session()``` by using ```clear_upon_failure``` this way ?

```python
        with openerp.api.Environment.manage():
            db = openerp.sql_db.db_connect(self.db_name)
            session = ConnectorSession(db.cursor(),
                                       self.uid,
                                       context=self.context)
            try:
                RegistryManager.check_registry_signaling(self.db_name)
                with session.clear_upon_failure():
                    yield session
                RegistryManager.signal_caches_change(self.db_name)
            except:
```